server/client_manager.py

```diff
@@ -324,4 +331,21 @@ def get_ooc_muted_clients(self):
         for client in self.clients:
             if client.is_ooc_muted:
                 clients.append(client)
-        return clients
\ No newline at end of file
+        return clients
+
+    def get_hdid_by_ip(self, ip):
+        hdid = None
+        for client in self.clients:
+            for client in self.clients:
+                if client.get_ip() == ip:
+                    return client.get_hdid()
+        if hdid == None:
+            return None
```

Comment:
Reaching this point in code assumes that no match was found. Therefore, the if check is not required.